File: apps/server/src/infrastructure/services/system-clock.ts
// src/infrastructure/services/system-clock.ts

import { Layer } from 'effect'
import { succeed } from 'effect/Effect'

import { Clock } from '../../ports/driven'

// ============================================
// SYSTEM CLOCK
// ============================================

export const SystemClockLive = Layer.succeed(Clock, {
  now: () => succeed(new Date()),
})

File: apps/server-java/src/main/java/com/maisonamane/infrastructure/service/SystemClock.java
package com.maisonamane.infrastructure.service;

import org.springframework.stereotype.Component;
import java.time.Instant;

/**
 * System clock provider.
 * Maps Effect-TS SystemClockLive to Spring @Component.
 */
@Component
public class SystemClock implements com.maisonamane.port.driven.service.Clock {

    @Override
    public Instant now() {
        return Instant.now();
    }
}
